### ISSUE TITLE
What's the difference between the two CLI for training？

### ISSUE THREAD

**Issue by autograd500** [ultralytics/ultralytics#5115]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

# Start training from a pretrained *.pt model
yolo detect train data=coco128.yaml model=yolov8n.pt epochs=100 imgsz=640

# Build a new model from YAML, transfer pretrained weights to it and start training
yolo detect train data=coco128.yaml model=yolov8n.yaml pretrained=yolov8n.pt epochs=100 imgsz=640

**Comment by glenn-jocher** [comment]
@autograd500 hello,

The main difference between these two command line interface (CLI) commands for training YOLOv8 lies in the initial model setup.

1. `yolo detect train data=coco128.yaml model=yolov8n.pt epochs=100 imgsz=640`: In this case, training starts with a pre-trained YOLOv8 model. We specify the .pt model file (in this case, `yolov8n.pt`), which means the weights from this pre-trained model are loaded and training starts from there. This is often used for fine-tuning a model on a specific dataset.

2. `yolo detect train data=coco128.yaml model=yolov8n.yaml pretrained=yolov8n.pt epochs=100 imgsz=640`: With this command, a new model architecture is built according to the specifications in the `yolov8n.yaml` file. Then, the weights from `yolov8n.pt` are transferred to this newly-created architecture, and only then does the training start. This is typically used when you aim to modify the structure of the original model and train on a specific dataset.

In summary, while the first command uses a pre-existing model and its weights, the second command creates a new model according to the architecture specified in `yolov8n.yaml` and transfers the weights from the pre-trained model to this new architecture.

Let us know if you need more information or other assistance!

**Comment by autograd500** [comment]
That is to say, if yolov8n.yaml in case 2 is the same as the original yolov8n.yaml, the 2 CLI commands are equal?

**Comment by glenn-jocher** [comment]
@autograd500 yes, you are correct! If the `yolov8n.yaml` used in the second command is indeed the same as the original architecture of the pre-trained `yolov8n.pt` model used in the first command, then the two CLI commands effectively perform the same action.

Keep in mind the difference is in the level of flexibility. The second command allows you to modify the model architecture as needed by editing the `yolov8n.yaml`, then load pre-trained weights into this modified architecture. However, if no modifications are made to the .yaml file and it matches the original model architecture, both commands will behave equivalently.

Let us know if this clears up your query or if there's anything else we can assist you with!
